crates/common/log_manager/src/log_utils.rs

```diff
@@ -172,95 +185,114 @@ mod tests {
+    /// file_d has timestamp: (current, not modified)
+    ///
+    /// The order of the output is { file_d, file_b }, because files are sorted from
+    /// most recent to oldest
+    fn test_filter_logs_path_containing_wildcard_on_type_and_metadata() {
         let (tempdir, files) = prepare();
         let tempdir_path = tempdir.path().to_str().unwrap();
 
-        let logs = filter_logs_on_type(&files, "type_one").unwrap();
+        let logs = filter_logs(&files, "type_one", datetime!(1970-01-01 00:00:03 +00:00)).unwrap();
 
         assert_eq!(
             logs,
             vec![
-                PathBuf::from(format!("{tempdir_path}/file_a")),
-                PathBuf::from(format!("{tempdir_path}/file_b")),
-                PathBuf::from(format!("{tempdir_path}/file_d"))
+                PathBuf::from(format!("{tempdir_path}/file_d_one")),
+                PathBuf::from(format!("{tempdir_path}/file_b_one")),
             ]
         )
     }
 
     #[test]
     /// Out of logs filtered on type = "type_one", that is: { file_a, file_b, file_d }.
-    /// Only logs filtered on metadata remain, that is { file_b, file_d }.
+    /// logs filtered on metadata are the same, that is { file_a, file_b, file_d }.
     ///
     /// This is because:
     ///
     /// file_a has timestamp: 1970/01/01 00:00:02
     /// file_b has timestamp: 1970/01/01 00:00:03
     /// file_d has timestamp: (current, not modified)
     ///
-    /// The order of the output is { file_d, file_b }, because files are sorted from
+    /// However, files are provided with static path so we don't take date range into consideration
+    ///
+    /// The order of the output is { file_d, file_b, file_a }, because files are sorted from
     /// most recent to oldest
-    fn test_filter_logs_path_on_metadata() {
-        let (tempdir, files) = prepare();
+    fn test_filter_logs_with_static_path_on_type_and_metadata() {
+        let (tempdir, _) = prepare();
         let tempdir_path = tempdir.path().to_str().unwrap();
 
-        let logs = filter_logs_on_type(&files, "type_one").unwrap();
-        let logs =
-            filter_logs_path_on_metadata("type_one", datetime!(1970-01-01 00:00:03 +00:00), logs)
-                .unwrap();
+        // provide vector of files with static paths
+        let files: Vec<FileEntry> = vec![
+            FileEntry {
+                path: format!("{tempdir_path}/file_a_one"),
+                config_type: "type_one".to_string(),
+            },
+            FileEntry {
+                path: format!("{tempdir_path}/file_b_one"),
+                config_type: "type_one".to_string(),
+            },
+            FileEntry {
+                path: format!("{tempdir_path}/file_c_two"),
+                config_type: "type_two".to_string(),
+            },
+            FileEntry {
+                path: format!("{tempdir_path}/file_d_one"),
+                config_type: "type_one".to_string(),
+            },
```

Comment:
I'm actually surprised that user can define multiple entries with the same `config_type`. But it makes sense. Good test case.